sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -600,14 +600,23 @@ public Symbol visitArrayComparisonExpression(ArrayComparisonExpression node, Exp
                 throw new UnsupportedFeatureException("ALL is not supported");
             }
 
-            Symbol leftSymbol = process(node.getLeft(), context);
-            Symbol arraySymbol = process(node.getRight(), context);
+            Expression left = node.getLeft();
+            Expression right = node.getRight();
+
+            if (right instanceof SubqueryExpression) {
+                context.registerSubqueryExpression((SubqueryExpression) right);
```

Comment:
i think it would be more straight forward to apply the single value function in this block after calling process on right, you have the symbol then and know what you need to do. also if applying the single value function you need to apply a limit of 2 to the select